### PR TITLE
Feature/question bank bulk upload

### PR DESCRIPTION
**Pull request by prashantesmagico**

### CODE REVIEW

File: src/file-upload/file-upload.controller.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          format : 'binary'
+        }
+      }
+    }
+  })
+  async uploadQuestions(@UploadedFile() file, @Res() res) {
+    // Log the initiaton for csv file upload
+    this.logger.log(`Initiate to upload a csv file.`);
+    try {
+      // Parsed the uploaded data
```
Comment (Ajay-Maury) [inline src/file-upload/file-upload.controller.ts]: implement all business logics in service and call a single method in controller

**Review by Ajay-Maury — CHANGES_REQUESTED** [review]

File: src/user-metadata/user-metadata.service.ts [hunk 2]
```diff
@@ -154,7 +154,7 @@ export class UserMetadataService {
                 Assessor: {
                     select:{
                         userId: true,
-                        userName: true,
+                        // userName: true,
```
Comment (VamshiBatta07) [inline src/user-metadata/user-metadata.service.ts]: Why is this commented out?

File: src/file-upload/file-upload.service.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import * as fs from "fs";
+import csvParser = require("csv-parser");
+
+@Injectable()
+export class FileUploadService {
+  constructor(private prisma: PrismaService) {}
+
+  public async storeData(data: CreateFileUploadDto[]) {
+    // Check if the competency Existed or not the given csv data if not then filter and throw error
+    const allCompetencies = await this.prisma.competency.findMany();
```
Comment (VamshiBatta07) [inline src/file-upload/file-upload.service.ts]: Please use the existing code in their respective services for db queries here and the other places.

File: src/file-upload/file-upload.service.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import { CreateFileUploadDto } from "./dto/create-file-upload.dto";
+import { PrismaService } from "src/prisma/prisma.service";
+import * as fs from "fs";
+import csvParser = require("csv-parser");
+
+@Injectable()
+export class FileUploadService {
+  constructor(private prisma: PrismaService) {}
+
+  public async storeData(data: CreateFileUploadDto[]) {
```
Comment (VamshiBatta07) [inline src/file-upload/file-upload.service.ts]: Please move this function to the question bank service.

**Review by VamshiBatta07 — CHANGES_REQUESTED** [review]

File: src/file-upload/file-upload.controller.ts (hunk 1 above)
Comment (prashantesmagico) [reply src/file-upload/file-upload.controller.ts]: Done

File: src/user-metadata/user-metadata.service.ts (hunk 2 above)
Comment (prashantesmagico) [reply src/user-metadata/user-metadata.service.ts]: Removed previously my server was not running

File: src/file-upload/file-upload.service.ts (hunk 3 above)
Comment (prashantesmagico) [reply src/file-upload/file-upload.service.ts]: modified the code by using the existing services

**Review by Ajay-Maury — APPROVED** [review]

**Review by VamshiBatta07 — APPROVED** [review]

**Pull request merged**